In [118]:
import numpy as np
import pandas as pd
import csv
pd.options.mode.chained_assignment = None  # default='warn'

In [119]:
from google.colab import drive
drive.mount('/content/drive')

Drive already mounted at /content/drive; to attempt to forcibly remount, call drive.mount("/content/drive", force_remount=True).


In [120]:
df = pd.read_csv('/content/drive/MyDrive/workspace/ProyectosPython/Dataset/gastos.csv', header=0)

In [121]:
df.columns

Index(['Gastos'], dtype='object')

In [122]:
dfangel = df.loc[df['Gastos'].str.contains('angel') & ~df['Gastos'].str.contains('kony') & ~df['Gastos'].str.contains('demi') & ~df['Gastos'].str.contains('debe')]
dfangel['valor'] = dfangel['Gastos'].str.extract('(\d+\.\d+|\d+)').astype(int)
dfangel

Gastos  valor
16                  Cena 390 angel;    390
17       Grab to airport 440 angel;    440
26             Repelente 155 angel;    155
29  Tuktuk a nuevo hostel 83 angel;     83
30               Hostel 2630 angel;   2630
31                 Tuktuk 80 angel;     80
32       Tour elefantes 6000 angel;   6000
35           Grab vuelta 100 angel;    100
41              Almuerzo 720 angel;    720
42               Mojitos 180 angel;    180
44       Pizza y cerveza 500 angel;    500
47        Comida papitas 160 angel;    160
48               7eleven 190 angel;      7
54                Desayu 140 angel;    140
60       Almuerzo railey 750 angel;    750
61                 Aguas 100 angel;    100
70              Cervezas 390 angel;    390

In [123]:
totalangel = dfangel['valor'].sum()
totalangel

12825

In [124]:
dfkony = df.loc[df['Gastos'].str.contains('kony') & ~df['Gastos'].str.contains('angel') & ~df['Gastos'].str.contains('demi') & ~df['Gastos'].str.contains('debe')]
dfkony['valor'] = dfkony['Gastos'].str.extract('(\d+\.\d+|\d+)').astype(int)
dfkony

Gastos  valor
0                 Grab 400 kony;    400
1                 Peaje 25 kony;     25
2                 Peaje 50 kony;     50
3     Rambutri hostel 4834 kony;   4834
6          Tuktuk mall 250 kony;    250
8               Comida 230 kony;    230
9                  Grab 92 kony;     92
10              Tuktuk 150 kony;    150
11        Comida float 430 kony;    430
14                Ferry kony 90;     90
15         Grb kony 164 ngel 10;    164
18        Taxi airport 150 kony;    150
19                Grab 117 kony;    117
20                Grab 123 kony;    123
25              90 kony revisar;     90
28              Tuktuk 120 kony;    120
34                 Grab 77 kony;     77
36                Grab 180 kony;    180
39                 Grab 80 kony;     80
51              Comida 860 kony;    860
52        Phipgi 4500 tour kony;   4500
55              Tragos 540 kony;    540
65  Entrada tiger cave 150 kony;    150
66                Grab 361 kony;    361
67                Grab 362 kony;    362
68              Helado 120 kony;    120
73                Grab 450 kony;    450

In [125]:
totalkony = dfkony['valor'].sum()
totalkony

14995

In [126]:
dfdemi = df.loc[df['Gastos'].str.contains('demi') & ~df['Gastos'].str.contains('angel') & ~df['Gastos'].str.contains('kony') & ~df['Gastos'].str.contains('debe')]
dfdemi['valor'] = dfdemi['Gastos'].str.extract('(\d+\.\d+|\d+)').astype(int)
dfdemi

Gastos  valor
22        Almuerzo 1500 demi;   1500
23     Alojamiento 2410 demi;   2410
24  Masajes y tour 5550 demi;   5550
33            Bebida 80 demi;     80
40   Hostel kaosok 1748 demi;   1748
58          Comida 1003 demi;   1003
64        Desayuno 1008 demi;   1008
69        Almuerzo 1170 demi;   1170
71             Cena 970 demi;    970
72        Desayuno 1003 demi;   1003

In [127]:
totaldemi = dfdemi['valor'].sum()
totaldemi

16442

In [128]:
dfpagos = df.loc[df['Gastos'].str.contains('debe')]
dfpagos['valor'] = dfpagos['Gastos'].str.extract('(\d+\.\d+|\d+)').astype(int)
dfpagos

Gastos  valor
38             Angel debe 200 demi;    200
45  Demi le debe 2500 a kony touri;   2500
49  Trekking angel debe 200 a kony;    200
59      Kony debe 200 angel ticket;    200

In [131]:
print(f"Total Angel: {totalangel}")
print(f"Total Kony: {totalkony}")
print(f"Total Demi: {totaldemi}")

tercio_angel = totalangel / 3
tercio_kony = totalkony / 3
tercio_demi = totaldemi / 3

print(f"Tercio Angel: {tercio_angel}")
print(f"Tercio Kony: {tercio_kony}")
print(f"Tercio Demi: {tercio_demi}")

Total Angel: 12825
Total Kony: 14995
Total Demi: 16442
Tercio Angel: 4275.0
Tercio Kony: 4998.333333333333
Tercio Demi: 5480.666666666667


1 - Angel debe a Kony 4998 - 4275 = 723

2 - Angel debe a Demi 5480 - 4275 = 1205

3 - Kony debe a Demi 5480 - 4998 = 482

# ahora sumamos el dfpagos

1 - Angel debe a Kony 4998 - 4275 = 723 + 200 -200 = **723**

2 - Angel debe a Demi 5480 - 4275 = 1205 + 200 = **1405**

** Kony debe a Demi 5480 - 4998 = 482 -2500= -2018**

3 -  ***Por ende Demi debe a Kony= 2018***
